jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -425,11 +425,22 @@ def _extract_from_aperture(self, spectral_cube, uncert_cube, aperture,
                                            uncertainty=collapsed_as_mean.uncertainty,
                                            wcs=collapsed_as_mean.wcs,
                                            meta=collapsed_as_mean.meta)
+        elif selected_func == 'sum':
+            collapsed_nddata = getattr(nddata_reshaped, selected_func)(
+                axis=spatial_axes, **kwargs
+            )  # returns an NDDataArray
+            # Remove per steradian denominator
+            if astropy.units.sr in collapsed_nddata.unit.bases:
```

Comment:
If you want to check that the unit is per solid angle (rather than just checking if the unit contains steradian), you can use core.validunits.check_if_unit_is_per_solid_angle. that would require some additional logic, which you could add to that function, to return what the solid angle unit is to multiply by.